modules/backend/visits/VisitCollection.hpp

```diff
@@ -1,25 +1,24 @@
 #pragma once
 
-#include "modules/backend/visits/visit/iVisit.hpp"
+#include <memory>
 #include <vector>
 
 namespace visit
 {
 
+class iVisit;
+
 class VisitCollection
 {
-    std::vector<iVisit> visits_;
-
   public:
-    void AppendVisit(const iVisit& visit)
-    {
-        visits_.emplace_back(visit);
-    }
+    void appendVisit(const std::shared_ptr<iVisit>& visit);
+    const std::vector<std::shared_ptr<iVisit>>& getVisits() const;
+
+  private:
+    //  TODO: type of pointer to be decided here (unique/shared/raw).
+    //  Functions require this consideration too then
+    std::vector<std::shared_ptr<iVisit>> visits_;
```

Comment:
Unique ptrs.
Raw pointers should be disallowed.
We need to add a rule, that raw pointers can be only used, when interface of external library has it, then it can be used, but still we should make it a safe pointer after obtaining such.

Reply:
Ok, it's good to decide early.
I will change to unique_ptr.

But I don't aggree to the rule exactly.
Because here by using unique_ptr we force on raw pointers and reference in our code as well.
From visit collection we will only return raw pointers and references then.
But this is good in my opinion: 
1) single point of managing resource
2) everywhere else in our code we can only use raw ptr or ref without bothering about object lifecycle.

Reply:
So we can add recommendation like this:
"where ever possible to establish single point of resource managment,use unique_ptr and return raw pointers or references from getters".
Or something like that